pkg/platform/provider/baremetal/validation/cluster.go

```diff
@@ -125,6 +128,8 @@ func ValidatClusterSpec(platformClient platformv1client.PlatformV1Interface, clu
 	allErrs = append(allErrs, ValidateClusterSpecVersion(platformClient, clusterName, cls.Spec.Version, fldPath.Child("version"), phase)...)
 	allErrs = append(allErrs, ValidateCIDRs(cls, fldPath)...)
 	allErrs = append(allErrs, ValidateClusterProperty(&cls.Spec, fldPath.Child("properties"))...)
+	allErrs = append(allErrs, ValidateCephFS(cls, fldPath)...)
```

Comment:
both ceph and nfs are storage, they should be merged in one validate item, like ValidateStorage on top level validate, then add ValidateCpehFS, ValidateNFS and other storage in feture in ValidateStorage.